### PR TITLE
Desk grid typescript

### PR DESCRIPTION
**Pull request by lbwexler**
Tom --

- There was one bit of suspicious code that I had to comment out in `CustomTabModels.ts`  See the TODO.
Seemed like it was not doing anything, but wanted to get you to have a look in case, what it was trying to do was actually needed.

- Also note a fix where `GridFindFieldImplModel.getValueGetters` was passing a Field instead of a string.  Not sure why we never ran into problems there, but that is fixed now.

Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [ ] Caught up with `develop` branch as of last change.
- [ ] Added CHANGELOG entry, or determined not required.
- [ ] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [ ] Updated doc comments / prop-types, or determined not required.
- [ ] Reviewed and tested on Mobile, or determined not required.
- [ ] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

File: desktop/cmp/rest/data/RestStore.ts [hunk 1]
```diff
@@ -4,44 +4,57 @@
...
 import '@xh/hoist/desktop/register';
 import {filter, keyBy, mapValues} from 'lodash';
-import {RestField} from './RestField';
+import {RestField, RestFieldSpec} from './RestField';
+
+
+export interface RestStoreConfig extends UrlStoreConfig {
+
+    /** Field names, configs, or instances. */
+    fields?: (string|RestFieldSpec|RestField)[];
```
Comment (TomTirapani) [inline desktop/cmp/rest/data/RestStore.ts:21]: Nice that we are able to type this override and pair it with defaultFieldClass. Should we also override fields on the RestStore class below to be typed as `fields: RestField[]`?

File: desktop/cmp/rest/data/RestStore.ts [hunk 2]
```diff
@@ -68,17 +81,17 @@ export class RestStore extends UrlStore {
         return resp;
     }
 
-    async addRecordAsync(rec) {
+    async addRecordAsync(rec: {id: string, data: PlainObject}) {
```
Comment (TomTirapani) [inline desktop/cmp/rest/data/RestStore.ts:84]: Should we type this as `UnsavedStoreRecordConfig` or similar?

Comment (TomTirapani) [reply desktop/cmp/rest/data/RestStore.ts:84]: I know the name is terrible lol

File: desktop/cmp/rest/impl/RestFormField.ts [hunk 3]
```diff
@@ -21,11 +21,6 @@ export const restFormField = hoistCmp.factory({
     displayName: 'RestFormField',
```
Comment (TomTirapani) [inline desktop/cmp/rest/impl/RestFormField.ts:21]: Should we add an @internal class doc to this component?

File: cmp/grid/GridModel.ts [hunk 4]
```diff
@@ -588,12 +595,20 @@ export class GridModel extends HoistModel {
         this.onCellClicked = onCellClicked;
         this.onCellDoubleClicked = onCellDoubleClicked;
         this.onCellContextMenu = onCellContextMenu;
+        this.appData = appData ? clone(appData) : {};
```
[the remark this answers is not in the document]
Comment (lbwexler) [reply cmp/grid/GridModel.ts:598]: I actually think that's overkill.  And who knows what someone will try to store here.

File: desktop/cmp/grid/editors/DateEditor.ts [hunk 5]
```diff
@@ -71,18 +71,19 @@ DateEditor.propTypes = {
...
  *
- * @argument {Object} data - The data object generated by the popper.js `update` method
- * @argument {Object} options - Modifiers configuration and options
- * @returns {Object} The data object, properly modified
+ * @argument data - The data object generated by the popper.js `update` method
+ * @argument options - Modifiers configuration and options
+ * @returns the data object, properly modified
  */
-function computeStyleInAgGrid(data, options, portalContainer) {
+function computeStyleInAgGrid(data: PlainObject, options: PlainObject, portalContainer): PlainObject {
```
[the remark this answers is not in the document]
Comment (lbwexler) [reply desktop/cmp/grid/editors/DateEditor.ts:78]: I tried that -- did not work --  but now I think it is an HTMLElement. -- it has `offsetHeight` and `scrollTop`.  Added that type and seems to look ok

File: desktop/cmp/grid/editors/EditorProps.ts [hunk 6]
```diff
@@ -4,21 +4,24 @@
...
+    gridModel: GridModel;
 
     /** StoreRecord being edited. */
-    record: PT.object,
+    record: StoreRecord;
 
     /** Props to pass through to the underlying HoistInput component */
-    inputProps: PT.object
-};
+    inputProps: DefaultHoistProps;
```
[the remark this answers is not in the document]
Comment (lbwexler) [reply desktop/cmp/grid/editors/EditorProps.ts:24]: Good question and I originally had that.  The problem is, that makes it tougher to specify a literal for the actual props you want, without casting it to `as HoistInputProps`

But you make a great point, and it's a shame to lose the type checking here.  

Seem like one option would be to make the editor props generic w.r.t. to the input props, will look at that!

File: desktop/cmp/grid/impl/filter/custom/CustomRowModel.ts [hunk 7]
```diff
@@ -5,29 +5,29 @@
...
+import {FieldFilterOperator, FieldFilterSpec} from '@xh/hoist/data';
+import {
+    ColumnHeaderFilterModel
+} from '@xh/hoist/desktop/cmp/grid/impl/filter/ColumnHeaderFilterModel';
 import {bindable, computed, makeObservable} from '@xh/hoist/mobx';
 import {isArray, isNil} from 'lodash';
+import { CustomTabModel } from './CustomTabModel';
 
 /**
  * @private
```
[the remark this answers is not in the document]
Comment (lbwexler) [reply desktop/cmp/grid/impl/filter/custom/CustomRowModel.ts:17]: done

File: desktop/cmp/grid/impl/filter/values/ValuesTabModel.ts [hunk 8]
```diff
@@ -4,42 +4,38 @@
...
  */
-import {GridAutosizeMode, GridModel} from '@xh/hoist/cmp/grid';
-import {HoistModel, managed, SizingMode} from '@xh/hoist/core';
-import {FieldType} from '@xh/hoist/data';
+import {GridModel} from '@xh/hoist/cmp/grid';
+import {HoistModel, managed} from '@xh/hoist/core';
+import {FieldFilterSpec} from '@xh/hoist/data';
+import {
+    ColumnHeaderFilterModel
+} from '@xh/hoist/desktop/cmp/grid/impl/filter/ColumnHeaderFilterModel';
```
[the remark this answers is not in the document]
Comment (lbwexler) [reply desktop/cmp/grid/impl/filter/values/ValuesTabModel.ts:12]: done

File: desktop/cmp/rest/data/RestStore.ts (hunk 2 above)
Comment (lbwexler) [reply desktop/cmp/rest/data/RestStore.ts:84]: yea, could be useful as a local type-- wondering if we need to export

File: desktop/cmp/rest/data/RestStore.ts (hunk 1 above)
Comment (lbwexler) [reply desktop/cmp/rest/data/RestStore.ts:21]: Done!  Needed to use 'declare' keyword, but it worked great.

File: desktop/cmp/rest/data/RestStore.ts (hunk 2 above)
Comment (lbwexler) [reply desktop/cmp/rest/data/RestStore.ts:84]: punting on this for now

**Pull request merged**